snapcraft/services/service_factory.py

```diff
@@ -21,37 +21,33 @@
 from dataclasses import dataclass
 from typing import TYPE_CHECKING
 
-from craft_application import ServiceFactory
+from craft_application import AppService, ServiceFactory
 
 from snapcraft import models, services
 
+# Add new services to this mapping to add them to the service factory
+_SERVICES: dict[str, type[AppService]] = {
+    "init": services.Init,
```

Comment:
If you take inspiration from craft_application's reset method about using class name strings and a string module name, snapcraft will be able to lazy-load the modules containing the services, speeding up startup.